extension/chrome/elements/compose.ts

```diff
@@ -260,6 +273,60 @@ export class ComposeView extends View {
   };
 
   public isCustomerUrlFesUsed = () => Boolean(this.fesUrl);
+
+  private preParseEmailRecipientsIfNeeded = async () => {
+    if (Catch.isThunderbirdMail() && this.thunderbirdMsgId && this.thunderbirdMsgId !== 0) {
+      const { headers, parts } = await browser.messages.getFull(this.thunderbirdMsgId);
```

Comment:
I noticed a couple issues when replying of forwarding message:
- when I replied to email message, it created a new thread instead of adding reply message to existing thread
- when I tried to forward message - recipients list was filled with recipients of original message, while it should be empty when forwarding.

I think we can simplify current implementation and just re-use existing code for populating data for reply and forward:
- there is `replyMsgId` url param which then used in `fetchReplyMeta` method for fetching reply data. You found a way how to get `msgId` from `thunderbirdMsgId` by using `headers?.['message-id'][0] || '';`, what about parsing it and using as `replyMsgId` param?
- there is `replyOption` url param which defines how recipients list should be populated, we can just convert `reply` and `forward` to `a_reply` and `a_forward`

Reply:
Hi @sosnovsky - thank you for raising this.

We have this limitation where the "message-id", in headers object only available to obtain when the secure compose click invoked from the `messageDisplay` and not from the `composeDisplay`. The place to identify if its part of `composeDisplay` is when the compose pop-up window from Thunderbird appears. To be very specific, the object message-id from headers is not present from `composeDisplay` which is sensible -- so there is that limitations with us where we can't rely on messageId to fetch threadId -> replyMsgId.

However, I can correct the reported behaviors and use the `replyOption` way of determining pre-filled data. I will proceed in this and let me know if the implementation I would be taking look good to you.

> when I replied to email message, it created a new thread instead of adding reply message to existing thread
> when I tried to forward message - recipients list was filled with recipients of original message, while it should be empty when forwarding.

Reply:
Will check and do the necessary changes to correct these. Thank you for your help.

Reply:
> when I replied to email message, it created a new thread instead of adding reply message to existing thread

@sosnovsky - I think I understand this now - when reply is clicked where an existing thread (previous email exchanges) are already made, then it should load and quote all of the conversation instead of just the current email where user replied to. I'll get it fixed.

Reply:
> when I replied to email message, it created a new thread instead of adding reply message to existing thread

To reproduce:
- open some thread in thunderbird
- click `reply`
- switch to `encrypted reply`
- compose and send reply from FlowCrypt encrypted compose
- then on recipient side sent reply will be on separate thread (not in the existing one)

Reply:
Got it - I see the bigger picture now on how it works. Will try to fix this and let you know.

Reply:
Thanks @sosnovsky - Its fixed. I think I have fixed it. Please feel free to perform additional checks as I might've them during my checkings.